### DISCUSSION TITLE
Following Ubuntu Install guide - "permission denied"

### DISCUSSION THREAD

**Discussion by quarfie** [requarks/wiki#4805] [Help / Questions] [answered]
I am following the instructions at https://docs.requarks.io/install/ubuntu on a new AWS t3a instance with Ubuntu 20.04.

When I got to this step:
```
# Create installation directory for Wiki.js
mkdir -p /etc/wiki
```
I got `mkdir: cannot create directory ‘/etc/wiki’: Permission denied`
so I repeated with sudo, but the next command:
```
openssl rand -base64 32 > /etc/wiki/.db-secret
```
also fails, returning `-bash: /etc/wiki/.db-secret: Permission denied` and this cannot be resolved with sudo

EDIT: I changed the owner of /etc/wiki with `sudo chown ubuntu:ubuntu /etc/wiki` and that allowed me to run the openssl command, but now I'm stuck with more permission problems at the next step:
```
docker network create wikinet
```
which is returning
```
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/networks/create": dial unix /var/run/docker.sock: connect: permission denied
```

**Comment by quarfie** [comment, marked as answer]
Solved!

Also needed to run `sudo usermod -a -G docker ubuntu` and then exit and make a new SSH connection.

This guide was very helpful in solving this: https://bobcares.com/blog/aws-ec2-docker-permission-denied/

**Comment by zak2002** [comment]
I am facing the same problem. I was wondering why choose /etc/wiki folder at all? why not create wiki folder in home directory to avoid permission troubles? Will it work? I am scared to try it as I don't want to mess up anything.
